mypy/build.py

```diff
@@ -265,6 +266,7 @@ def _build(sources: List[BuildSource],
                            fscache=fscache)
 
     try:
+        TypeState.reset_all()
```

Comment:
If we aren't running a fine-grained incremental build, I think that we can reset the state after the build (before returning the build results), since it won't be useful any more at than point but it will consume some memory. We should also reset the cache when shutting down the daemon, maybe in `reset()`. It's good manners to not leave global state lying around at the end of a program.

Style nit: Move this to be outside the try statement, since we don't expect this to fail.